tests/test_tokenization.py

```diff
@@ -9,12 +15,303 @@ def _assert_tokenization(tokenizer: TokenizerWrapper):
+            ),
+            15,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 15,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=False,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            15,
+            0,
+        ),
+        # Test cases 3: Sequence is longer than max model input length, i.e., len(text) > max model input length
+        # NOTE: This is a typical case when tokenising the pretraining dataset!!!
+        (
+            "AAAAAAAA" * 1030,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=False,
+                padding=False,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1030,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1030,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=False,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1030,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1030,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1024,
+            0,
+        ),
+        # if we want to pad to max model input length we have to use padding="max_length",
+        # otherwise we will only pad to the longest sequence in the batch.
+        # see: https://huggingface.co/docs/transformers/pad_truncation#padding-and-truncation
+        (
+            "AAAAAAAA" * 1020,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding=True,
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1020,
+            0,
+        ),
+        (
+            "AAAAAAAA" * 1020,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=True,
+                padding="max_length",
+                max_length=None,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            1024,
+            4,
+        ),
+    ],
+)
+def test_hf_tokenize(
+    text: str,
+    tokenizer_config: PreTrainedHFTokenizerConfig,
+    expected_length: int,
+    expected_num_paddding_tokens: int,
```

Comment:
```suggestion
    expected_num_padding_tokens: int,
```
Typo, occurrence 1 of 3